File: js/chess-knight.js
function chessKnight(cell){
    let board = { 
    'a': 1,
    'b': 2,
    'c': 3,
    'd': 4,
    'e': 5,
    'f': 6,
    'g': 7,
    'h': 8   
    }
    
    let knightX = board[cell[0]]
    let knightY = Number(cell[1])
    
    let possibleKnightX = [knightX - 2, knightX - 1, knightX + 1, knightX + 2] 
    let possibleKnightY = [knightY - 2, knightY - 1, knightY + 1, knightY + 2] 
    
    for(let i = 0 ; i < possibleKnightX.length; i++){
        if(possibleKnightX[i] <= 0 || possibleKnightX[i] > 8) possibleKnightX[i] = NaN
        
        if(possibleKnightY[i] <= 0 || possibleKnightY[i] > 8) possibleKnightY[i] = NaN
    }
        
    let movements = [
    possibleKnightX[0] + possibleKnightY[1],
    possibleKnightX[0] + possibleKnightY[2],
    possibleKnightX[1] + possibleKnightY[0],
    possibleKnightX[1] + possibleKnightY[3],
    possibleKnightX[2] + possibleKnightY[0],
    possibleKnightX[2] + possibleKnightY[3],
    possibleKnightX[3] + possibleKnightY[1],
    possibleKnightX[3] + possibleKnightY[2]
    ]
    
    let actualMovements = movements.filter(e => !isNaN(e))
    
    return actualMovements.length
}

let cell = "a1"

console.log(`For a knight located in the cell ${cell} of a standard chess board there could be`, chessKnight(cell), `valid movements`)

File: go/chess-knight/chess-knight.go
package main

import(
	"fmt"
)

func chessKnight(cell string) int{

	knightX, knightY := int(rune(cell[0])), int(rune(cell[1]))

	possibleKnightX, possibleKnightY := []int{knightX - 2, knightX - 1, knightX + 1, knightX + 2}, []int{knightY - 2, knightY - 1, knightY + 1, knightY + 2}

	for idx, val := range possibleKnightX{
		if val < 97 || val > 104{
			possibleKnightX[idx] = -1
		}
		if possibleKnightY[idx] < 49 || possibleKnightY[idx] > 56{
			possibleKnightY[idx] = -1
		}

	}

	movements := []int{
	possibleKnightX[0] + possibleKnightY[1],
    possibleKnightX[0] + possibleKnightY[2],
    possibleKnightX[1] + possibleKnightY[0],
    possibleKnightX[1] + possibleKnightY[3],
    possibleKnightX[2] + possibleKnightY[0],
    possibleKnightX[2] + possibleKnightY[3],
    possibleKnightX[3] + possibleKnightY[1],
    possibleKnightX[3] + possibleKnightY[2]}
	
	validMovements := 0

	for _, val := range movements{
		if val >= 146 && val <= 160{
			validMovements++
		}
	}

	return validMovements
}

func main(){
	cell := "a1"

	fmt.Printf("For a knight located in the cell %s of a standard chess board there could be %d valid movements", cell, chessKnight(cell))
}